src/HE.Investment.AHP.Domain/HomeTypes/ValueObjects/DecimalValueObject.cs

```
@@ -0,0 +1,86 @@
+using System.Globalization;
+using HE.Investments.Common.Domain;
+using HE.Investments.Common.Extensions;
+using HE.Investments.Common.Messages;
+using HE.Investments.Common.Validators;
+
+namespace HE.Investment.AHP.Domain.HomeTypes.ValueObjects;
+
+public abstract class DecimalValueObject : ValueObject
+{
+    protected DecimalValueObject(
+        string? value,
+        string fieldName,
+        string displayName,
+        bool isCalculation,
+        decimal minValue = decimal.MinValue,
+        decimal maxValue = decimal.MaxValue)
+    {
+        if (value.IsNotProvided() && !isCalculation)
+        {
+            return;
```

In this case, we should throw an exception. Otherwise object will be created with default values

---

done